code/drasil-example/dblpendulum/app/Main.hs

```diff
@@ -13,6 +13,6 @@ main = do
   dumpEverything fullSI ".drasil/"
   typeCheckSI fullSI
   gen (DocSpec (docChoices SRS [HTML, TeX, JSON]) "DblPendulum_SRS") srs printSetting
-  genCode choices code
+  genCode choices code "../../../drasil-metadata/lib/Data/Drasil/DrasilConfiguration.json"
```

Comment:
Maybe we can use template Haskell to inject the contents of the file inside a variable inside `drasil-metadata` instead of referring to the file URI and passing around a file URI + re-reading it for each project, when it's a static thing.

Reply:
Yes, that is a good idea. I will make that change right now.

Reply:
While not exactly what I had in mind for a solution, it looks like dfdcbca33d6fe9c1b8ed8cff17b443948c49e441 does the trick. Since the "3" is a special number, though, we might as well have it hard-coded in the path too (and not passed around as an argument by things). Have you made sure that the path works fine for in the example subprojects too?

Reply:
When you say subprojects, you are referring to the projectile example, right? If we are talking about the projectile example, the only location where I am adding a number for the `genCode` is in the `Choices.hs` file. In that, I am putting in 4 instead of 3.

Reply:
My guess is that this number will be added to some sort of record at some point (similarly to the number used for getting the file path of the Drasil logo)

Reply:
Currently, it is not added to any record as the `genCode` function is not a record and this number is only being used at a very early stage that it is not part of a `DrasilState` or `CodeSpec` or any of those major datasets. We do not have a record for this right now as this would be the only member of the record.  I am not adding it into the `DrasilConfig` record because that is created specifically for obtaining the data in the `DrasilConfiguration.json` file.

Reply:
Agreed!

Reply:
Is what I have done in 95a5942d57b294c91f16e968f73a8a46d3f97829 what you were talking about or am I misunderstanding it @balacij?

Reply:
I was meaning that we could/should use [TemplateHaskell](http://wiki.haskell.org/Template_Haskell) to embed the file contents at Drasils compile-time. I made an [MVP demo](https://github.com/balacij/mvp-TemplateHaskellReadFile/blob/main/src/Lib.hs) that you can use to try it out, basically just wrapping a [stackoverflow answer](https://stackoverflow.com/a/44369564/16760741) in a Stack project.

I'm not sure if we want to go down this route because it's TemplateHaskell, but I think it's better than passing around a number related to the current working directory and the version json file.

Pinging @JacquesCarette for his thoughts here, since he was noted the `Maybe Int` in a later comment.

Reply:
Yes, a more sophisticated implementation would use TH. We already use TH, so it's not a big deal to use it some more.

It does feel like maybe the first implementation could be simpler than using TH. Even with TH, we'd need to pass information around, and it shouldn't be a new argument, so the `Maybe Int` design issue is independent of that.

Reply:
> Even with TH, we'd need to pass information around, and it shouldn't be a new argument, so the Maybe Int design issue is independent of that.

I think that if we computed the TH in `drasil-metadata`, and referenced `drasil-metadata` everywhere else, we _shouldn't_ (hopefully) need to play the "what is the folder depth?"/`Maybe Int` game.

Reply:
Good point - I agree with that assessment.

So, maybe it should be done now? Can you help @harmanpreet-sagar with that?

Reply:
I have created a new PR #3566 to resolve #3501. In that while not explicitly taking in any input for the folder depth, I used the number that we have specified in the `Choices` regardless for the `README` generations and modified it slightly to generate the length as I determined a pattern for that. This way, we are not taking in any input for the specific folder depth nor do we have any `Maybe Int` issues. 

I unfortunately was unable to determine how to use TH in `drasil-metadata` so I used another method. Thus, if we would like to implement that, I might need a little bit of assistance in using TH.

Reply:
Let's try out the TH approach, I think it will be easiest to work with once we have it working. Did you have a look at the prototype I linked @harmanpreet-sagar ? I'll message you on Discord too.